types/event.go

```diff
@@ -92,7 +114,46 @@ func (t *Trace) GetSamplerKey() (string, bool) {
 // Span is an event that shows up with a trace ID, so will be part of a Trace
 type Span struct {
 	Event
-	TraceID string
+	TraceID     string
+	DataSize    int
+	ArrivalTime time.Time
+}
+
+// GetDataSize computes the size of the Data element of the Span.
+// Note that it's not the full size of the span, but we're mainly using this for
+// relative ordering, not absolute calculations.
+func (sp *Span) GetDataSize() int {
+	total := 0
+	// the data types we should be getting from JSON are:
+	// float64, int64, bool, string
+	for _, v := range sp.Data {
+		switch v.(type) {
+		case bool:
+			total += 1
+		case float64, int64, int:
+			total += 8
+		case string, []byte:
+			total += len(v.(string))
+		default:
+			total += 8 // catchall
+		}
+	}
+	return total
+}
+
+// cacheImpactFactor controls how much we overweight older spans compared to newer ones;
+// setting this to 1 means they're not weighted by duration
+const cacheImpactFactor = 4
+
+// CacheImpact calculates an abstract value for something we're calling cache impact, which is
+// the product of the size of the span and a factor related to the amount of time the span
+// has been stored in the cache, based on the TraceTimeout value.
+func (sp *Span) CacheImpact(traceTimeout time.Duration) int {
+	// multiplier will be a value from 1-cacheImpactFactor, depending on how long the
+	// span has been in the cache compared to the traceTimeout.
+	multiplier := int(cacheImpactFactor*time.Since(sp.ArrivalTime)/traceTimeout) + 1
```

Comment:
Are we assuming that the time since arrival can never exceed the trace timeout?

Reply:
Not necessarily, and this won't break if it does, but I've documented a bit better. Thanks for the question.